ConwaysGameOfLife/GameController.cs

```diff
@@ -0,0 +1,112 @@
+using System;
+
+namespace ConwaysGameOfLife
+{
+    public class GameController
+    {
+        // responsible for implementing game
+        // orchestrator type class - brings all together
+
+        /*
+            welcome message
+            user asked for universe dimensions
+            dimensions parsed
+            universe initialized
+            initial universe printed to console
+            user asked for location of alive cells
+            input parsed/validated
+            universe set up with live cells
+            universe printed
+            game plays
+            rules checked
+            new universe regenerated
+            ends either by user typing q or all dead cells
+
+            input
+            output
+            universe
+            rules
+        */
+        private Universe _universe;
+
+        private IOutput _output;
+
+        private IInput _input;
+
+        public GameController(Universe universe, IOutput output, IInput input)
+        {
+            _universe = universe;
+            _output = output;
+            _input = input;
+        }
+
+        public void CreateInitialUniverse()
+        {
+            _output.WriteLine(Messages.RequestDimensions);
+            var input = CreateValidUniverseString();
+            InputParser.ParseUniverse(input);
+            DisplayUniverse();

```

Comment:
Displaying the created `Universe` may be considered a "side-effect" i.e. the purpose of the function would be fulfilled without actually displaying the `Universe`. Consider explicitly calling `DisplayUniverse` when needed in the `Run` function.
```suggestion
```